tests/std/tests/P0896R4_views_iota/test.cpp

```diff
@@ -239,6 +241,57 @@ constexpr void test_integral() {
     }
 }
 
+template <class W>
+constexpr void test_one_difference() {
+    ranges::iota_view r{numeric_limits<W>::min(), numeric_limits<W>::max()};
+    using Diff = conditional_t<sizeof(W) >= sizeof(long long), _Signed128,
+        conditional_t<sizeof(W) >= sizeof(int), long long, int>>;
+    static_assert(same_as<decltype(r.end() - r.begin()), Diff>);
+    static_assert(noexcept(r.end() - r.begin()));
+    const auto n = (Diff{1} << (numeric_limits<W>::digits + (signed_integral<W> ? 1 : 0))) - 1;
+    assert(r.end() - r.begin() == n); // left > right
+    assert(r.begin() - r.end() == -n); // right > left
+}
+
+constexpr bool test_difference() {
+    // Ensure we have full coverage of all branches in `operator-(i, s)`
+
+    // signed integer-like, sizeof(W) < sizeof(int)
+    test_one_difference<signed char>();
+    test_one_difference<short>();
+
+    // unsigned integer-like, sizeof(W) < sizeof(int)
+    test_one_difference<unsigned char>();
+    test_one_difference<unsigned short>();
+
+    // signed integer-like, sizeof<int> <= sizeof(W) < sizeof(long long)
+    test_one_difference<int>();
+
+    // unsigned integer-like, sizeof<int> <= sizeof(W) < sizeof(long long)
+    test_one_difference<unsigned int>();
+
+    // signed integer-like, sizeof<long long> <= sizeof(W)
+    test_one_difference<long long>();
+
+    // unsigned integer-like, sizeof<long long> <= sizeof(W)
+    test_one_difference<unsigned long long>();
+
+    // non-integer-like
+    {
+        char* some_chars = new char[1ull << 24];
+        ranges::iota_view r{&some_chars[0], &some_chars[1ull << 24]};
+        using Diff = ptrdiff_t;
+        static_assert(same_as<decltype(r.end() - r.begin()), Diff>);
+        static_assert(noexcept(r.end() - r.begin()));
+        const auto n = Diff{1} << 24;

```

Comment:
Suggestion: Make `constexpr`, move (along with `Diff`) up and replace the existing uses of `1ull << 24` so we don't need to repeat the magic number.